### PR TITLE
optimize task metrics collection

### PR DESCRIPTION
**Pull request by rangermeier**
- remove tasks from self._state.tasks once they reach one of the READY_STATES (SUCCESS, FAILURE, REVOKED)
- use .inc() for task gauge with READY_STATES labels because they can't go down
- don't use separate event listeners for `task-received` and `task-started`
- don't use a separate dict to gather task latency information

### CODE REVIEW

File: celery_prometheus_exporter.py [hunk 1]
```diff
@@ -48,26 +49,40 @@ def _process_event(self, evt):
...
-    def _collect_tasks(self, evt):
-        self._state._event(evt)
+    def _observe_latency(self, evt):
+        try:
+            prev_evt = self._state.tasks[evt['uuid']]
+        except KeyError:
+            pass
+        else:
+            # ignore latency if it is a retry
+            if prev_evt.state == celery.states.RECEIVED:
```
Comment (zerok) [inline celery_prometheus_exporter.py]: If the KeyError actually happens, wouldn't that check result in an NameError due to `prev_evt` not being defined?

Comment (rangermeier) [inline celery_prometheus_exporter.py]: the `else:`-block is only executed when there is no exception in the `try:`-block, so it should be save.

Comment (zerok) [inline celery_prometheus_exporter.py]: Argh, I totally overlooked the `else`. Sorry about that :)

**Comment by rangermeier** [comment]
Hi, I rebased on master and fixed compatibility with Celery 3 - an issue I discovered thanks to the new tests :)

**Comment by zerok** [comment]
Thank you :)

**Pull request merged**